src/transformers/training_args.py

```diff
@@ -531,6 +536,10 @@ class TrainingArguments:
     push_to_hub: bool = field(
         default=False, metadata={"help": "Whether or not to upload the trained model to the model hub after training."}
     )
+    resume_from_checkpoint: Optional[str] = field(
+        default=None,
+        metadata={"help": "The path to a folder with a valid checkpoint for your model."},
```

Comment:
```suggestion
        metadata={"help": "The path to a folder with a valid checkpoint for your model. It's not used by the Trainer directly."},
```

If someone looks at help and not at the docs, let's signal that it's not a real Trainer arg.

But why didn't we do the same for max sample args? these fall into the same category. 

This makes me think of having a subset of `TrainingArgs` that are there for the scripts but not used by the Trainer?

Reply:
`max_samples` is an argument linked to the data, so goes in `DataArguments`. This is an argument linked to the training, so goes into `TrainingArgument`, like `do_train`, `do_eval` or `do_predict`.
If someone is looking at the help, they are looking at a training example, so there is no need for the mention there IMO (same way it's done for `do_train`, `do_eval` or `do_predict`).

Reply:
> max_samples is an argument linked to the data, so goes in DataArguments. 

Except we don't have `DataArguments` - I was proposing to create it. 

> This is an argument linked to the training, so goes into TrainingArgument, like do_train, do_eval or do_predict.

Oh, I see. `TrainingArguments` already contains non-Trainer-training arguments, that's fair. I was just proposing to have these in a separate group.

We don't have to discuss this in this PR, but as more args are being added that aren't `Trainer` it'd be good to group those and split them away, rather then needing to write disclaimers like:
```
This argument is not directly used by :class:`~transformers.Trainer`, it's
            intended to be used by your training/evaluation scripts instead. See the `example scripts
            <https://github.com/huggingface/transformers/tree/master/examples>`__ for more details.
```

Plus I proposed not to replicate the same args again and again in every script, when they can all be listed once in a nice class. e.g. you proposed `DataArguments`

So we could for example have `script_args` which would contain `ScriptDataArguments` and `ScriptTrainingArguments` and leave `TrainingArguments` to only contain args that `Trainer`-specific.

This is an idea.

Reply:
Removing any field of `TrainingArguments` to put it elsewhere would break BC, so the split would have to be handled carefully. Renaming the `TrainingArguments` would be to disruptive, even for a major version release, so that name is probably going to stick. We can look at `ScriptTrainingArguments`/`ScriptDataArguments` however.

In any case, this is definitely for another PR :-)

Reply:
right, I keep forgetting BC :(